### ISSUE TITLE
Saving models with ONNX. 

### ISSUE THREAD

**Issue by Joao-L-S-Almeida** [terrastackai/terratorch#323]
Is something interesting for TerraTorch to have a way to export models to ONNX format ?

**Comment by rbavery** [comment]
Yes! ONNX or Pytorch ExportedPrograms are more portable than distributing the model with weights files and code, no need to install a complex project to use the model, just onnx runtime or torch. And in the export/compile process you can get speedups and reduced gpu memory during inference.  

Between ONNX and ExportedProgram, I think ExportedProgram might be better to start with. I think the speedups might be more significant. and ExportedPrograms have the simplest dependencies, just Pytorch to load and run the model.

https://pytorch.org/docs/stable/export.html#an-example

**Comment by Joao-L-S-Almeida** [comment]
Hi, @rbavery Thank you for the feedback and for the example.

**Comment by romeokienzler** [comment]
@Joao-L-S-Almeida let me create a PR for the ONNX exporter
